pipelines/pipedrive/custom_fields_munger.py

```diff
@@ -0,0 +1,56 @@
+from dlt.common.normalizers.names.snake_case import normalize_column_name
+from typing import Dict, List, Generator, Optional
+
+import dlt
+
+
+def munge_push_func(data: List, endpoint: str) -> List:
+    """
+    Specific function to perform data munging and push changes to custom fields' mapping stored in dlt's state
+    The endpoint must be an entity fields' endpoint
+    """
+    custom_fields_mapping = dlt.state().setdefault('custom_fields_mapping', {})
+    for data_item in data:
+        if all([data_item.get('edit_flag'), data_item.get('name'), data_item.get('key')]):
+            if custom_fields_mapping.get(endpoint):
+                if data_item['key'] not in custom_fields_mapping[endpoint]:
```

Comment:
I do not understand this line. what `key` does? why are we skipping the update if it exists? also you should update the mapping  in `data_item_mapping` with mapping from state to preserve the original values in state and just add new ones form `data_item_mapping`

Reply:
Regarding custom fields, "key" field contains pipedrive's hash string representation of its name.
I assumed that pipedrive's hash strings are meant to be a univoque representation of custom fields names (per endpoint), so dlt's state shouldn't be updated if those values remained unchanged.
First of all I search on/update/add values on dlt's state, and then I end up updating data with dlt's state values.